### PR TITLE
EOS-19071 : Change the s3 provisioner config file to include phase names

### PR DESCRIPTION
**Pull request by shahidSTX**
Change the s3 prov config file to include phase names.

Files modified:
   scripts/provisioning/configcmd.py
   scripts/provisioning/s3_haproxy_config.py
   scripts/provisioning/s3_prov_config.yaml
   scripts/provisioning/setupcmd.py

Testing: 
   Single node provisioning all phases done - `post_install`, `prepare`, `config`, `init`.

Signed-off-by: Mohammed Shahid <mohammed.shahid@seagate.com>

### CODE REVIEW

**Review by amitwac2608 — APPROVED** [review]

File: scripts/provisioning/s3_prov_config.yaml [hunk 1]
```diff
@@ -20,15 +20,19 @@
...
-CONFSTORE_CLUSTER_ID_KEY: "server_node>{}>cluster_id"                               # server_node>{machine-id}>cluster_id
-CONFSTORE_S3INSTANCES_KEY: "server_node>{}>s3_instances"                            # server_node>{machine-id}>s3_instances
-CONFSTORE_HOSTNAME_KEY: "server_node>{}>hostname"                                   # server_node>{machine-id}>hostname
-CONFSTORE_SITE_COUNT_KEY: "cluster>{}>site_count"                                   # cluster>{cluster-id}>site_count
-CONFSTORE_STORAGE_SET_COUNT_KEY: "cluster>{}>site>storage_set_count"                # cluster>{cluster-id}>site>storage_set_count
-CONFSTORE_STORAGE_SET_SERVER_NODES_KEY: "cluster>{}>storage_set[{}]>server_nodes"   # cluster>{cluster-id}>storage_set[N]>server_nodes
-CONFSTORE_PUBLIC_FQDN_KEY: server_node>{}>network>data>public_fqdn                  # server_node>{machine-id}>network>data>public_fqdn
-CONFSTORE_PRIVATE_FQDN_KEY: server_node>{}>network>data>private_fqdn                # server_node>{machine-id}>network>data>private_fqdn
+CONFSTORE_SITE_COUNT_KEY: "cluster>{}>site_count"                                     # cluster>{cluster-id}>site_count
```
Comment (nileshgovande) [inline scripts/provisioning/s3_prov_config.yaml:23]: Add test, reset and cleanup phase

Comment (shahidSTX) [reply scripts/provisioning/s3_prov_config.yaml:23]: Done.

File: scripts/provisioning/s3_prov_config.yaml [hunk 2]
```diff
@@ -20,15 +20,22 @@
...
+POST_INSTALL:
+PREPARE:
+CONFIG:
```
Comment (nileshgovande) [inline scripts/provisioning/s3_prov_config.yaml:26]: 1)Replace value of key with stuff from comment and change all .format to .replace
2)We agree on inheritance hirarchy
3)Move unsolicited keys down the file and add a note/guideline at the start of the file which will ask to add all unsolicited keys below

Comment (shahidSTX) [reply scripts/provisioning/s3_prov_config.yaml:35]: Done.

File: scripts/provisioning/s3_prov_config.yaml [hunk 3]
```diff
@@ -16,19 +16,43 @@
...
-CONFSTORE_ROOTDN_PASSWD_KEY: "cortx>software>openldap>root>secret"
-CONFSTORE_CLUSTER_ID_KEY: "server_node>{}>cluster_id"                               # server_node>{machine-id}>cluster_id
-CONFSTORE_S3INSTANCES_KEY: "server_node>{}>s3_instances"                            # server_node>{machine-id}>s3_instances
-CONFSTORE_HOSTNAME_KEY: "server_node>{}>hostname"                                   # server_node>{machine-id}>hostname
-CONFSTORE_SITE_COUNT_KEY: "cluster>{}>site_count"                                   # cluster>{cluster-id}>site_count
-CONFSTORE_STORAGE_SET_COUNT_KEY: "cluster>{}>site>storage_set_count"                # cluster>{cluster-id}>site>storage_set_count
-CONFSTORE_STORAGE_SET_SERVER_NODES_KEY: "cluster>{}>storage_set[{}]>server_nodes"   # cluster>{cluster-id}>storage_set[N]>server_nodes
-CONFSTORE_PUBLIC_FQDN_KEY: server_node>{}>network>data>public_fqdn                  # server_node>{machine-id}>network>data>public_fqdn
-CONFSTORE_PRIVATE_FQDN_KEY: server_node>{}>network>data>private_fqdn                # server_node>{machine-id}>network>data>private_fqdn
+CONFSTORE_SITE_COUNT_KEY: "cluster>cluster-id>site_count"
```
Comment (nileshgovande) [inline scripts/provisioning/s3_prov_config.yaml:58]: Is it an unsolicited key ? I mean not a part of provisioner keystore ?

Comment (shahidSTX) [reply scripts/provisioning/s3_prov_config.yaml:58]: Yes, no phases are directly using this key.

**Review by AjinkyaDhumal — APPROVED** [review]

**Pull request merged**